db/models.py

```diff
@@ -50,3 +55,49 @@
 
     def __str__(self) -> str:
         return f"{self.movie.title} {str(self.show_time)}"
+
+
+class Order(models.Model):
+    user = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE)
+    created_at = models.DateTimeField(auto_now_add=True)
+
+    class Meta:
+        ordering = ["-created_at"]
+
+    def __str__(self) -> str:
+        return f"Order: {self.created_at}"
+
+
+class Ticket(models.Model):
+    movie_session = models.ForeignKey(to=MovieSession, on_delete=models.CASCADE)
+    order = models.ForeignKey(to=Order, on_delete=models.CASCADE)
+    row = models.IntegerField()
+    seat = models.IntegerField()
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                fields=["movie_session", "seat", "row"],
+                name="unique_ticket",
+            )
+        ]
+
+    def clean(self):
+        if self.row > self.movie_session.cinema_hall.rows:
+            raise ValidationError({"row": ["Row number must be in available range."]})
+        if self.seat > self.movie_session.cinema_hall.rows:
+            raise ValidationError({"seat": ["Seat number must be in available range."]})
```

Comment:
The validation logic for `seat` is incorrect. It should compare `self.seat` with `self.movie_session.cinema_hall.seats_in_row` instead of `self.movie_session.cinema_hall.rows`.